wcivf/apps/people/tests/test_person_views.py

```diff
@@ -195,44 +201,57 @@ def test_previous_party_affiliations(self):
             post=post,
             ballot_paper_id="local.welsh.assembly.2022-05-01",
         )
-        PersonPostFactory(
+        past_pe = PostElectionFactory(
+            election=past_election,
+            post=post,
+            ballot_paper_id="local.welsh.assembly.2021-05-01",
+        )
+
+        person_post = PersonPostFactory(
             person=self.person,
             election=election,
             party=party,
             party_name=party.party_name,
             post=post,
             post_election=pe,
         )
-
-        past_election = ElectionFactory(
-            name="Welsh Assembly Election",
-            current=False,
-            election_date="2021-05-01",
-            slug="local.welsh.assembly.2021-05-01",
-        )
         past_person_post = PersonPostFactory(
             person=self.person,
             election=past_election,
             party=party,
+            party_name=party.party_name,
             post=post,
-            post_election=pe,
+            post_election=past_pe,
         )
 
         previous_party_1 = PartyFactory(
             party_name="Conservative", party_id="foo_id"
         )
         previous_party_2 = PartyFactory(party_name="Labour", party_id="bar_id")
-        past_person_post.previous_party_affiliations.add(previous_party_1)
-        past_person_post.previous_party_affiliations.add(previous_party_2)
-
+        person_post.previous_party_affiliations.add(previous_party_1)
+        person_post.previous_party_affiliations.add(previous_party_2)
+        self.assertEqual(person_post.previous_party_affiliations.count(), 2)
+        self.assertEqual(self.person.past_not_current_candidacies.count(), 1)
+        self.assertEqual(self.person.current_or_future_candidacies.count(), 1)
         response = self.client.get(self.person.get_absolute_url(), follow=True)
         self.assertTemplateUsed(
             "people/includes/_person_previous_elections_card.html"
         )
+        self.assertTemplateUsed(
+            "elections/includes/_previous_party_affiliations.html"
+        )
+        self.assertTemplateUsed("people/includes/_person_intro_card.html")
         self.assertContains(
-            response, "Other party affiliations in the past 12 months"
+            response,
+            "Other party affiliations in the past 12 months",
+            html=True,
         )
         self.assertContains(response, "Conservative, Labour", html=True)
+        self.assertContains(
+            response,
+            "has declared their affiliation with the following parties in the past 12 months",
+            html=True,
+        )
```

Comment:
This turns up when I inspect `response.content` 😕

Reply:
When you check the `response.content` does it have any extra whitespaces or line breaks etc?

Reply:
It has line breaks. In the past, setting `html=True` has handled this. 
`response.content` includes `<p>\n    \n        Candidate 29 has declared their affiliation with the following parties in the past 12 months:\n        \n            Conservative, \n        \n            Labour\n        \n    \n</p>` so I thought this would pass regardless.

Reply:
That's really odd as you are right it looks like it would pass fine... could try reducing text?

Reply:
Actually I just tried running this locally without the `html=True` and it passed. But dont ask me why 🤷

Reply:
Weird. Ok thanks for trying it. Will remove.